samples/Mvc.RenderViewToString/RazorViewHelper.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.AspNet.Http.Internal;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.Abstractions;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Mvc.ViewEngines;
+using Microsoft.AspNet.Mvc.ViewFeatures;
+using Microsoft.AspNet.Routing;
+
+namespace Mvc.RenderViewToString
+{
+    public class RazorViewHelper
+    {
+        private ICompositeViewEngine _viewEngine;
+        private ITempDataProvider _tempDataProvider;
+        private IServiceProvider _serviceProvider;
+
+        public RazorViewHelper(
+            ICompositeViewEngine viewEngine,
+            ITempDataProvider tempDataProvider,
+            IServiceProvider serviceProvider)
+        {
+            _viewEngine = viewEngine;
+            _tempDataProvider = tempDataProvider;
+            _serviceProvider = serviceProvider;
+        }
+
+        private ActionContext GetActionContext()
+        {
+            var httpContext = new DefaultHttpContext();
+            httpContext.RequestServices = _serviceProvider;
+            return new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
+        }
+
+        public string RenderViewToString<TModel>(string name, TModel model)
+        {
+            var actionContext = GetActionContext();
+
+            var viewEngineResult = _viewEngine.FindView(actionContext, name, false);
+
+            if (!viewEngineResult.Success)
+            {
+                return "View not found";
```

Comment:
This looks too much like success to the caller. Either `bool TryRenderViewToString(string, TModel, out string)` or `throw` here.